src/shared/api/dto.ts

```diff
@@ -1,183 +1,136 @@
-    metadata: any
-    /**
-     * amount of triggers, always 0 for now
-     */
-    triggers: number
-  }
-
-  interface PublicKey {
-    digest_function: string
-    payload: string
-  }
-
-  interface Status {
-    peers: string
-    blocks: string
-    txs: string
-    uptime: {
-      secs: string
-      nanos: string
-    }
-  }
-
-  export interface BlockShallow {
-    height: number
-    /**
-     * ISO DateTime
-     */
-    timestamp: string
-    block_hash: string
-    /**
-     * Transactions count
-     */
-    transactions: number
-    /**
-     * Rejected transactions count
-     */
-    rejected_transactions: number
-  }
-
-  export interface Block {
-    height: number
-    /**
-     * See {@link BlockShallow.timestamp}
-     */
-    timestamp: string
-    block_hash: string
-    parent_block_hash: string
-    rejected_transactions_merkle_root_hash: string
-    invalidated_blocks_hashes: string[]
-    /**
-     * List of serialized {@link @iroha2/data-model#VersionedValidTransaction}
-     */
-    transactions: string[]
-    /**
-     * List of serialized {@link @iroha2/data-model#VersionedRejectedTransaction}
-     */
-    rejected_transactions: string[]
-    /**
-     * List of hashes. WIP always empty
-     */
-    view_change_proofs: string[]
-  }
-
-  export interface TransactionDto {
-    /**
-     * WIP zeroed
-     */
-    hash: string
-    block_hash: string
-    // TODO: bloch_height is missing from backend response
-    block_height: number
-    payload: TransactionPayload
-    signatures: Signature[]
-    rejection_reason?: string
-  }
-
-  export interface TransactionPayload {
-    account_id: string
-    instructions: TransactionInstructions
-    /**
-     * ISO timestamp
-     */
-    creation_time: string
-    time_to_live_ms: number | null
-    nonce: number | null
-    metadata: any
-  }
-
-  /**
-   * `Instructions` `string[]` - list of serialized
-   * {@link @iroha2/data-model#Instruction}
-   */
-  export type TransactionInstructions = Tagged<'Instructions', string[]> | Tagged<'Wasm', undefined>;
-
-  export interface Signature {
-    /**
-     * Public key's multihash
-     */
-    public_key: string
-    /**
-     * Hex binary
-     */
-    payload: string
-  }
+import { z } from 'zod';
+
+const paginationSchema = z.object({
+  page: z.number(),
+  per_page: z.number(),
+  // TODO: remove nullable when backend is ready
+  total_pages: z.number().nullable(),
+  total_items: z.number().nullable(),
+});
+
+export type Pagination = z.infer<typeof paginationSchema>;
+
+export interface Paginated<T> {
+  pagination: Pagination
+  items: T[]
 }
+
+const paginationParamsSchema = z
+  .object({
+    page: z.number(),
+    per_page: z.number(),
+  })
+  .partial();
+
+export type PaginationParams = z.infer<typeof paginationParamsSchema>;
+
+export const accountSchema = z.object({
+  id: z.string(),
+  metadata: z.record(z.string(), z.any()),
+});
+
+export type AccountDto = z.infer<typeof accountSchema>;
```

Comment:
DTO usually desribe the data that comes from the API directly. If it is transformed to something else on frontend, `Dto` suffix seems extra.

Zod allows combining the definition of DTO with its transformation to a convenient type.

```suggestion
export type Account = z.infer<typeof accountSchema>;
```